kubetest/kind/kind.go

```diff
@@ -383,11 +393,25 @@ func (d *Deployer) DumpClusterLogs(localPath, gcsPath string) error {
 // TestSetup is a NO-OP in this deployer.
 func (d *Deployer) TestSetup() error {
 	log.Println("kind.go:TestSetup()")
-	// set KUBECONFIG, but ignore the error if a cluster is not up yet.
-	_ = d.setKubeConfigEnv()
 
 	// set conformance env so ginkgo.sh etc won't try to do provider setup
 	os.Setenv("KUBERNETES_CONFORMANCE_TEST", "y")
+
+	// Proceed only if a cluster exists.
+	exists, err := d.clusterExists()
+	if err != nil {
+		return err
+	}
+	if !exists {
+		log.Printf("kind.go:TestSetup(): no such cluster %q; skipping the setup of KUBECONFIG!", d.kindClusterName)
+		return nil
+	}
+
+	// set KUBECONFIG, but ignore the error if a cluster is not up yet.
```

Comment:
the second part of this comment seems wrong, we ignore the error by way of never reaching this code

Reply:
oops, updated.